Client/Frontend/Widgets/PhotonActionSheetProtocol.swift

```diff
@@ -159,6 +161,28 @@ extension PhotonActionSheetProtocol {
             }
         }
         
+        let pinToTopSites = PhotonActionSheetItem(title: Strings.PinTopsiteActionTitle, iconString: "action_pin") { action in
+            guard let url = tab.url?.displayURL else { return }
+            let absoluteString = url.absoluteString
+            
+            let sql = self.profile.favicons as! SQLiteHistory
+            sql.getSitesForURLs([absoluteString]).uponQueue(.main) { result in
+                guard let results = result.successValue else {
+                    log.warning("Could not get site for \(absoluteString)")
+                    return
+                }
+                
+                if case let site?? = results[0] {
```

Comment:
curious about the syntax here `if case let site?? = results[0]`, this looks like it would still crash if 0 was out of bounds. you could do a `let site = results.first` instead